certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/validation/model/ElementValidationDiagnosis.java

```diff
@@ -0,0 +1,129 @@
+package se.inera.intyg.certificateservice.domain.validation.model;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Value;
+import se.inera.intyg.certificateservice.domain.certificate.model.ElementData;
+import se.inera.intyg.certificateservice.domain.certificate.model.ElementValue;
+import se.inera.intyg.certificateservice.domain.certificate.model.ElementValueDiagnosis;
+import se.inera.intyg.certificateservice.domain.certificate.model.ElementValueDiagnosisList;
+import se.inera.intyg.certificateservice.domain.certificatemodel.model.ElementId;
+import se.inera.intyg.certificateservice.domain.certificatemodel.model.FieldId;
+import se.inera.intyg.certificateservice.domain.diagnosiscode.model.DiagnosisCode;
+import se.inera.intyg.certificateservice.domain.diagnosiscode.repository.DiagnosisCodeRepository;
+
+@Value
+@Getter(AccessLevel.NONE)
+@Builder
+public class ElementValidationDiagnosis implements ElementValidation {
+
+  private static final String DIAGNOSIS_DESCRIPTION = ".description";
+  FieldId mandatoryField;
+  List<FieldId> order;
+  DiagnosisCodeRepository diagnosisCodeRepository;
+
+  @Override
+  public List<ValidationError> validate(ElementData data,
```

Comment:
I think it would add readability if this method would be refactored into several methods

Reply:
I made some efforts to try to improve the readability, let me know if you think it was good enought.

Im not super satisfied with the order handling, maybe we need to come up with a different solution, or what you think?

Reply:
After discussion, we added a todo and will handle it later.

Reply:
After discussion, we decided that this is good enought for now.